bundle/tests/registered_model/databricks.yml

```diff
@@ -0,0 +1,32 @@
+resources:
+  registered_models:
+    my_registered_model:
+      name: "my-model"
+      comment: "comment"
+      catalog_name: "main"
+      schema_name: "default"
+      grants:
+        - privileges:
+            - EXECUTE
+          principal: "account users"
```

Comment:
(I'm not sure what happened to the other thread but let me start a new one just in case.)

I wonder if this should look like

```
grants:
  - level: EXECUTE
    group_name: users
```

that syntax would make it more symmetrical to permissions, which would be less surprising to DAB users. It would also make the difference between users, groups, and principals more explicit. In the UC syntax, that difference is not made which means that group names with an `@` cannot be supported and that friendly names of service principals cannot be supported either.

One argument against the syntax above could be that there's already a well-established REST API for dealing with Grants. (I couldn't find it.) We generally tend to follow REST APIs for our YML design.

cc @shreyas-goenka @pietern

Reply:
Note we use the `/unity-catalog/permissions/function` API for model grants. I could not find RestAPI docs for this as well.

The rest API docs are probably the same as the terraform attributes, but indeed should be verified.
https://github.com/databricks/terraform-provider-databricks/blob/master/docs/resources/grants.md#metastore-grants

Reply:
API docs are available at https://docs.databricks.com/api/workspace/grants/get.

They indeed talk about principals and privileges rather than `<user/group/service_principal>_name` and levels.

Reply:
Note the current approach is indeed faithful to the API.

Reply:
To fully close on this thread: there are some interesting trade-offs here. A key DABs design guideline has been to follow the API. Here, the API doesn't actually have a "grants" property at the resource level, but we are at least applying the same signature for grants as seen elsewhere. But then while we gain consistency there, we lose on consistency with the permissions signatures.

What I suggest as a follow-up is to review our validation (once we have proper validation). We should show reasonable errors whenever a user tries to do

```
grants:
  - level: EXECUTE
    group_name: users
```

or

```
permissions:
        - privileges:
            - CAN_WRITE
          principal: lennart@databricks.com
```